astropy/modeling/projections.py

```diff
@@ -1963,18 +1963,25 @@ def evaluate(cls, x, y, matrix, translation):
         x, y : array, float
               x and y coordinates
         """
+        if hasattr(x, 'unit') and hasattr(y, 'unit'):
```

Comment:
I was trying to make this not quantity specific. (I wonder if there should be a unit ABC that all unit packages could use...)